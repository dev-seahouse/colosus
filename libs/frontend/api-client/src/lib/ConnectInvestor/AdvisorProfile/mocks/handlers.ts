import { rest } from 'msw';
import type { ConnectInvestorGetAdvisorProfileResponseDto } from '../AdvisorProfile';

const BASE_URL =
  'http://localhost:9000/api/v1/connect/investor/advisor-profile';
export const connectInvestorAdvisorProfileApiHandlers = [
  rest.get(BASE_URL, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json<ConnectInvestorGetAdvisorProfileResponseDto>({
        canPerformTransactActions: true,
        advisorSubscriptionIds: 'CONNECT',
        advisorSubscriptionInPlace: true,
        jobTitle: 'Financial Planner',
        incomeThreshold: 100000,
        retireeSavingsThreshold: 100000,
        firstName: 'Matius',
        lastName: 'Ben Slayer',
        tenantRealm: 'matius_at_bambu_co',
        profileBioRichText:
          '<p>As a financial planner, my primary objective is to help my clients achieve clarity and peace of mind in achieving their financial goals. I prioritize investments that focus on generating income, followed by long-term growth, to help you build a solid financial foundation for your future.</p>',
        contactMeReasonsRichText:
          "<p><strong>Let's make it happe</strong>n.</p><p></p><p>Hi there, I'm Robert Henderson, CFP.</p><p>Let's work together to bridge the financial gap and achieve your goals.</p><ul><li><p>Offer diversified investment strategies to minimize risk and optimise the return.</p></li><li><p>Extensive tax planning knowledge to minimize your tax liability .</p></li></ul>",
      })
    );
  }),
];
