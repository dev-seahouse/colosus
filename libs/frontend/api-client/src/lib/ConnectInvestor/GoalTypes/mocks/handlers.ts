import { rest } from 'msw';
import type { ConnectInvestorGetGoalTypesResponseDto } from '../GoalTypes';

const BASE_URL = 'http://localhost:9000/api/v1/connect/investor/goal-types';
export const connectInvestorGoalTypesApiHandlers = [
  rest.get(BASE_URL, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json<ConnectInvestorGetGoalTypesResponseDto>({
        goalTypes: [
          {
            id: '8d299530-c622-47cb-b425-82429d7443c0',
            name: 'Retirement',
            description: 'Retire comfortably',
          },
          {
            id: 'd0f191ec-5ddc-4f18-81c6-38b89ac7b5a6',
            name: 'House',
            description: 'Buy a house',
          },
          {
            id: '03c46eb2-8004-4fb9-b400-ec3e9903b0db',
            name: 'Education',
            description: 'Save for college fees',
          },
          {
            id: 'b4dd784c-1743-4161-8b9e-d6e460b6c6c4',
            name: 'Growing Wealth',
            description: 'Just want to grow my wealth',
          },
          {
            id: 'a4d7abd5-dcbc-44b7-86ca-a0b0cbbd3465',
            name: 'Other',
            description: 'I have another goal in mind',
          },
        ],
      })
    );
  }),
];
