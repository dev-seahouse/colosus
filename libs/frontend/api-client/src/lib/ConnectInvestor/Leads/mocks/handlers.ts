import { rest } from 'msw';

const BASE_URL = 'http://localhost:9000/api/v1/connect/investor/leads';

export const connectInvestorLeadsApiHandler = [
  rest.post(BASE_URL, (req, res, ctx) => {
    return res(ctx.status(200));
  }),
];
