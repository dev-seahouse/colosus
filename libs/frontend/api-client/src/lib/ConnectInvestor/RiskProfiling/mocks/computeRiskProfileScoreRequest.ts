import { ConnectInvestorComputeRiskProfileScoreRequestDto } from '../RiskProfiling';

export const r = {
  questionnaireId: 'cbacafa5-7d7e-4d39-9bc3-6ce1839e3266',
  questionnaireVersion: 1,
  answers: [
    {
      questionGroupingId: '7ca3ad84-fa07-4bbf-8e5e-d21f5347df47',
      questionId: 'c0b51a4b-4f17-42c9-8e63-2fa78dc52dc3',
      answerId: '',
      answerScoreNumber: 21,
    },
    {
      questionGroupingId: '7ca3ad84-fa07-4bbf-8e5e-d21f5347df47',
      questionId: '3b216868-890a-4c47-9566-f5b6c983bd71',
      answerId: '417e9e23-88b7-4563-98e5-4638dacf4154',
      answerScoreNumber: null,
    },
    {
      questionGroupingId: '422e5300-286f-4294-aa4b-b7adcc273bb2',
      questionId: 'b8d89946-6f33-4b77-8ea5-e49484421890',
      answerId: '',
      answerScoreNumber: 10,
    },
    {
      questionGroupingId: 'cb5bcc9e-a911-4be0-834b-50c9e00902f8',
      questionId: 'a624ebe3-ae95-4581-97d7-4edcae4c3f77',
      answerId: '',
      answerScoreNumber: 30,
    },
    {
      questionGroupingId: '18e3b245-8138-4e01-85b8-0dd71c549e1e',
      questionId: 'd5fe0921-a78f-42d7-8305-bf6014e0325e',
      answerId: '1fd21a14-0bb5-40e6-9564-4178fb5461b8',
      answerScoreNumber: null,
    },
    {
      questionGroupingId: '73ef469b-0ce4-4094-b7aa-9eb1fba08c76',
      questionId: 'ef508bec-c453-47f2-9d1d-e77b10409fb6',
      answerId: '01f8d423-1d4d-42fd-8e42-91310a40c8d7',
      answerScoreNumber: null,
    },
    {
      questionGroupingId: '8bed6ec9-b81f-4190-8ba7-d4f2ed4aea97',
      questionId: 'c22a1a56-ed1b-47c6-bda6-461c125ac0d4',
      answerId: 'c35126f0-9526-43da-bab7-83d9bad6546d',
      answerScoreNumber: null,
    },
    {
      questionGroupingId: '8bed6ec9-b81f-4190-8ba7-d4f2ed4aea97',
      questionId: 'a5b99ad4-3e58-40ad-a114-5589d54c8b85',
      answerId: 'e92ee43e-776b-44eb-83e9-c4edc43bf9cf',
      answerScoreNumber: null,
    },
  ],
} satisfies ConnectInvestorComputeRiskProfileScoreRequestDto;
