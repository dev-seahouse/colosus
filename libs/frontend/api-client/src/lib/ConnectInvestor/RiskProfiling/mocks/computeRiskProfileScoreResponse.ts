import { ConnectInvestorComputeRiskProfileScoreResponseDto } from '../RiskProfiling';

export const r = {
  questionnaireId: 'cbacafa5-7d7e-4d39-9bc3-6ce1839e3266',
  questionnaireVersion: 1,
  questionnairePayload: [
    {
      questionGroupingId: '7ca3ad84-fa07-4bbf-8e5e-d21f5347df47',
      questionId: 'c0b51a4b-4f17-42c9-8e63-2fa78dc52dc3',
      answerId: '',
      answerScoreNumber: 21,
    },
    {
      questionGroupingId: '7ca3ad84-fa07-4bbf-8e5e-d21f5347df47',
      questionId: '3b216868-890a-4c47-9566-f5b6c983bd71',
      answerId: '417e9e23-88b7-4563-98e5-4638dacf4154',
      answerScoreNumber: null,
    },
    {
      questionGroupingId: '422e5300-286f-4294-aa4b-b7adcc273bb2',
      questionId: 'b8d89946-6f33-4b77-8ea5-e49484421890',
      answerId: '',
      answerScoreNumber: 10,
    },
    {
      questionGroupingId: 'cb5bcc9e-a911-4be0-834b-50c9e00902f8',
      questionId: 'a624ebe3-ae95-4581-97d7-4edcae4c3f77',
      answerId: '',
      answerScoreNumber: 30,
    },
    {
      questionGroupingId: '18e3b245-8138-4e01-85b8-0dd71c549e1e',
      questionId: 'd5fe0921-a78f-42d7-8305-bf6014e0325e',
      answerId: '1fd21a14-0bb5-40e6-9564-4178fb5461b8',
      answerScoreNumber: null,
    },
    {
      questionGroupingId: '73ef469b-0ce4-4094-b7aa-9eb1fba08c76',
      questionId: 'ef508bec-c453-47f2-9d1d-e77b10409fb6',
      answerId: '01f8d423-1d4d-42fd-8e42-91310a40c8d7',
      answerScoreNumber: null,
    },
    {
      questionGroupingId: '8bed6ec9-b81f-4190-8ba7-d4f2ed4aea97',
      questionId: 'c22a1a56-ed1b-47c6-bda6-461c125ac0d4',
      answerId: 'c35126f0-9526-43da-bab7-83d9bad6546d',
      answerScoreNumber: null,
    },
    {
      questionGroupingId: '8bed6ec9-b81f-4190-8ba7-d4f2ed4aea97',
      questionId: 'a5b99ad4-3e58-40ad-a114-5589d54c8b85',
      answerId: 'e92ee43e-776b-44eb-83e9-c4edc43bf9cf',
      answerScoreNumber: null,
    },
  ],
  riskProfileComputationResult: [
    {
      questionCategoryName: 'Risk Capacity',
      questionId: '1',
      questionScoreDetails: [
        {
          questionCategoryName: 'FINANCIAL_HEALTH',
          questionId: '1.1',
          questionScoreDetails: [
            {
              questionCategoryName: 'INCOME',
              questionId: '1',
              questionScore: '5',
              questionWeight: '0.5',
            },
            {
              questionCategoryName: 'BALANCE_SHEET',
              questionId: '2',
              questionScore: '1',
              questionWeight: '0.5',
            },
          ],
          questionScore: '5',
          questionWeight: '0.4',
        },
        {
          questionCategoryName: 'AGE',
          questionId: '1.2',
          questionScore: '5',
          questionWeight: '0.2',
        },
        {
          questionCategoryName: 'GOAL',
          questionId: '1.3',
          questionScore: '4',
          questionWeight: '0.2',
        },
        {
          questionCategoryName: 'FINANCIAL_KNOWLEDGE',
          questionId: '1.4',
          questionScore: '1',
          questionWeight: '0.2',
        },
      ],
      questionScore: '4',
      questionWeight: '0.5',
    },
    {
      questionCategoryName: 'Risk Tolerance',
      questionId: '2',
      questionScoreDetails: [
        {
          questionCategoryName: 'RISK_COMFORT_LEVEL',
          questionId: '1',
          questionScore: '2.33333333',
          questionWeight: '0.5',
        },
        {
          questionCategoryName: 'RISK_COMFORT_LEVEL',
          questionId: '2',
          questionScore: '2.33333333',
          questionWeight: '0.5',
        },
      ],
      questionScore: '2',
      questionWeight: '0.5',
    },
  ],
  riskProfileScore: '2',
  modelRiskProfile: {
    riskProfileId: '095f203e-ef0c-42fe-8e17-eecdf8ab69b7',
    riskProfileName: 'Conservative',
    riskProfileDescription:
      'You are OK with a bit of volatility in your portfolio.<br/>You understand that the value of your portfolio may go down for a short period of time before it bounces back.<br/>You understand that expected returns are below average.',
  },
  modelPortfolioId: '3d3dd80b-a899-4650-b9e0-588cc8fb1664',
} satisfies ConnectInvestorComputeRiskProfileScoreResponseDto;
