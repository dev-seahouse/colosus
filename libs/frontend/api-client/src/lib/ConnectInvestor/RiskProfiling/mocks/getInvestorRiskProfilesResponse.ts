import { ConnectInvestorGetInvestorRiskProfilesResponseDto } from '../RiskProfiling';

export const r = [
  {
    id: '8e6fb5fd-e8b4-4019-9d2c-6fd1ff8f6e5f',
    lowerLimit: '1',
    upperLimit: '1',
    riskProfileName: 'Very Conservative',
    riskProfileDescription:
      'You don’t want to experience volatility in your portfolio.<br/>You don’t expect the value of your portfolio to go down at any significant period of time.<br/>You understand that expected returns are low',
    tenantId: '0bc1e3e3-63fb-4495-beb8-411054baecda',
  },
  {
    id: 'eeef55b0-40a7-4980-b8ec-a94e01be1d13',
    lowerLimit: '2',
    upperLimit: '2',
    riskProfileName: 'Conservative',
    riskProfileDescription:
      'You are OK with a bit of volatility in your portfolio.<br/>You understand that the value of your portfolio may go down for a short period of time before it bounces back.<br/>You understand that expected returns are below average.',
    tenantId: '0bc1e3e3-63fb-4495-beb8-411054baecda',
  },
  {
    id: '4bf6874c-e5ea-4f66-b2f9-e1212b823422',
    lowerLimit: '3',
    upperLimit: '3',
    riskProfileName: 'Balanced',
    riskProfileDescription:
      'You are OK with some volatility in your portfolio.<br/>You understand that the value of your portfolio may go down for a moderate period of time before it bounces back.<br/>You understand that expected returns are average.',
    tenantId: '0bc1e3e3-63fb-4495-beb8-411054baecda',
  },
  {
    id: 'e39f6a89-743f-4ed2-9307-22d5d545020c',
    lowerLimit: '4',
    upperLimit: '4',
    riskProfileName: 'Growth',
    riskProfileDescription:
      'You are OK with volatility in your portfolio.<br/>You understand that the value of your portfolio may go down for a significant period of time before it bounces back.<br/>You expect good returns in the mid to long term.',
    tenantId: '0bc1e3e3-63fb-4495-beb8-411054baecda',
  },
  {
    id: '149a15e6-7152-4a2d-9d16-0d6a0c70b825',
    lowerLimit: '5',
    upperLimit: '5',
    riskProfileName: 'Aggressive',
    riskProfileDescription:
      'You are OK with high volatility in your portfolio.<br/>You understand that the value of your portfolio may go down sharply in the future but you know that you will reap big benefits if you are patient enough.<br/>You expect high returns in the long term.',
    tenantId: '0bc1e3e3-63fb-4495-beb8-411054baecda',
  },
] satisfies ConnectInvestorGetInvestorRiskProfilesResponseDto;
