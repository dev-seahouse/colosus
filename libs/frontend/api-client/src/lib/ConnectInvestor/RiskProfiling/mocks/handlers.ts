import { rest } from 'msw';

import { r as questionnaireResponse } from './getRiskQuestionnaireResponse';
import { r as computeRiskProfileScoreResponse } from './computeRiskProfileScoreResponse';
import { r as getInvestorRiskProfilesResponse } from './getInvestorRiskProfilesResponse';

import {
  ConnectInvestorGetInvestorRiskProfilesResponseDto,
  ConnectInvestorComputeRiskProfileScoreResponseDto,
  ConnectGetInvestorRiskQuestionnaireResponseDto,
} from '../RiskProfiling';

const BASE_URL = 'http://localhost:9000/api/v1/connect/investor/risk-profiling';

export const connectInvestorRiskProfilingHandlers = [
  rest.get(`${BASE_URL}/questionnaire`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json<ConnectGetInvestorRiskQuestionnaireResponseDto>(
        questionnaireResponse
      )
    );
  }),

  rest.post(`${BASE_URL}/compute`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json<ConnectInvestorComputeRiskProfileScoreResponseDto>(
        computeRiskProfileScoreResponse
      )
    );
  }),

  rest.get(`${BASE_URL}/risk-profiles`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json<ConnectInvestorGetInvestorRiskProfilesResponseDto>(
        getInvestorRiskProfilesResponse
      )
    );
  }),
];
