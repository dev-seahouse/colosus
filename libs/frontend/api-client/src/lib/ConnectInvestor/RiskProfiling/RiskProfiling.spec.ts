import ConnectInvesorRiskProfilingApi from './RiskProfiling';
import { r as computeRiskProfileScoreReuqest } from './mocks/computeRiskProfileScoreRequest';

describe('RiskProfiling', () => {
  const connectInvesorRiskProfilingApi = new ConnectInvesorRiskProfilingApi();

  describe('getRiskProfileQuesetionnaire', () => {
    it('should return correct response', async () => {
      const res =
        await connectInvesorRiskProfilingApi.getInvestorRiskQuestionnaire();
      expect(res.data).toMatchSnapshot();
    });
  });

  describe('computeRiskProfileScore', () => {
    it('should return correct response', async () => {
      const res = await connectInvesorRiskProfilingApi.computeRiskProfileScore(
        computeRiskProfileScoreReuqest
      );
      expect(res.data).toMatchSnapshot();
    });
  });

  describe('getInvestorRiskProfiles', () => {
    it('should return correct response', async () => {
      const res =
        await connectInvesorRiskProfilingApi.getInvestorRiskProfiles();
      expect(res.data).toMatchSnapshot();
    });
  });
});
