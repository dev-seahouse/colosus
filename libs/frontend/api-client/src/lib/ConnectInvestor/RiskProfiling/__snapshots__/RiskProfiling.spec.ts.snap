// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RiskProfiling > computeRiskProfileScore > should return correct response 1`] = `
{
  "modelPortfolioId": "3d3dd80b-a899-4650-b9e0-588cc8fb1664",
  "modelRiskProfile": {
    "riskProfileDescription": "You are OK with a bit of volatility in your portfolio.<br/>You understand that the value of your portfolio may go down for a short period of time before it bounces back.<br/>You understand that expected returns are below average.",
    "riskProfileId": "095f203e-ef0c-42fe-8e17-eecdf8ab69b7",
    "riskProfileName": "Conservative",
  },
  "questionnaireId": "cbacafa5-7d7e-4d39-9bc3-6ce1839e3266",
  "questionnairePayload": [
    {
      "answerId": "",
      "answerScoreNumber": 21,
      "questionGroupingId": "7ca3ad84-fa07-4bbf-8e5e-d21f5347df47",
      "questionId": "c0b51a4b-4f17-42c9-8e63-2fa78dc52dc3",
    },
    {
      "answerId": "417e9e23-88b7-4563-98e5-4638dacf4154",
      "answerScoreNumber": null,
      "questionGroupingId": "7ca3ad84-fa07-4bbf-8e5e-d21f5347df47",
      "questionId": "3b216868-890a-4c47-9566-f5b6c983bd71",
    },
    {
      "answerId": "",
      "answerScoreNumber": 10,
      "questionGroupingId": "422e5300-286f-4294-aa4b-b7adcc273bb2",
      "questionId": "b8d89946-6f33-4b77-8ea5-e49484421890",
    },
    {
      "answerId": "",
      "answerScoreNumber": 30,
      "questionGroupingId": "cb5bcc9e-a911-4be0-834b-50c9e00902f8",
      "questionId": "a624ebe3-ae95-4581-97d7-4edcae4c3f77",
    },
    {
      "answerId": "1fd21a14-0bb5-40e6-9564-4178fb5461b8",
      "answerScoreNumber": null,
      "questionGroupingId": "18e3b245-8138-4e01-85b8-0dd71c549e1e",
      "questionId": "d5fe0921-a78f-42d7-8305-bf6014e0325e",
    },
    {
      "answerId": "01f8d423-1d4d-42fd-8e42-91310a40c8d7",
      "answerScoreNumber": null,
      "questionGroupingId": "73ef469b-0ce4-4094-b7aa-9eb1fba08c76",
      "questionId": "ef508bec-c453-47f2-9d1d-e77b10409fb6",
    },
    {
      "answerId": "c35126f0-9526-43da-bab7-83d9bad6546d",
      "answerScoreNumber": null,
      "questionGroupingId": "8bed6ec9-b81f-4190-8ba7-d4f2ed4aea97",
      "questionId": "c22a1a56-ed1b-47c6-bda6-461c125ac0d4",
    },
    {
      "answerId": "e92ee43e-776b-44eb-83e9-c4edc43bf9cf",
      "answerScoreNumber": null,
      "questionGroupingId": "8bed6ec9-b81f-4190-8ba7-d4f2ed4aea97",
      "questionId": "a5b99ad4-3e58-40ad-a114-5589d54c8b85",
    },
  ],
  "questionnaireVersion": 1,
  "riskProfileComputationResult": [
    {
      "questionCategoryName": "Risk Capacity",
      "questionId": "1",
      "questionScore": "4",
      "questionScoreDetails": [
        {
          "questionCategoryName": "FINANCIAL_HEALTH",
          "questionId": "1.1",
          "questionScore": "5",
          "questionScoreDetails": [
            {
              "questionCategoryName": "INCOME",
              "questionId": "1",
              "questionScore": "5",
              "questionWeight": "0.5",
            },
            {
              "questionCategoryName": "BALANCE_SHEET",
              "questionId": "2",
              "questionScore": "1",
              "questionWeight": "0.5",
            },
          ],
          "questionWeight": "0.4",
        },
        {
          "questionCategoryName": "AGE",
          "questionId": "1.2",
          "questionScore": "5",
          "questionWeight": "0.2",
        },
        {
          "questionCategoryName": "GOAL",
          "questionId": "1.3",
          "questionScore": "4",
          "questionWeight": "0.2",
        },
        {
          "questionCategoryName": "FINANCIAL_KNOWLEDGE",
          "questionId": "1.4",
          "questionScore": "1",
          "questionWeight": "0.2",
        },
      ],
      "questionWeight": "0.5",
    },
    {
      "questionCategoryName": "Risk Tolerance",
      "questionId": "2",
      "questionScore": "2",
      "questionScoreDetails": [
        {
          "questionCategoryName": "RISK_COMFORT_LEVEL",
          "questionId": "1",
          "questionScore": "2.33333333",
          "questionWeight": "0.5",
        },
        {
          "questionCategoryName": "RISK_COMFORT_LEVEL",
          "questionId": "2",
          "questionScore": "2.33333333",
          "questionWeight": "0.5",
        },
      ],
      "questionWeight": "0.5",
    },
  ],
  "riskProfileScore": "2",
}
`;

exports[`RiskProfiling > getInvestorRiskProfiles > should return correct response 1`] = `
[
  {
    "id": "8e6fb5fd-e8b4-4019-9d2c-6fd1ff8f6e5f",
    "lowerLimit": "1",
    "riskProfileDescription": "You don’t want to experience volatility in your portfolio.<br/>You don’t expect the value of your portfolio to go down at any significant period of time.<br/>You understand that expected returns are low",
    "riskProfileName": "Very Conservative",
    "tenantId": "0bc1e3e3-63fb-4495-beb8-411054baecda",
    "upperLimit": "1",
  },
  {
    "id": "eeef55b0-40a7-4980-b8ec-a94e01be1d13",
    "lowerLimit": "2",
    "riskProfileDescription": "You are OK with a bit of volatility in your portfolio.<br/>You understand that the value of your portfolio may go down for a short period of time before it bounces back.<br/>You understand that expected returns are below average.",
    "riskProfileName": "Conservative",
    "tenantId": "0bc1e3e3-63fb-4495-beb8-411054baecda",
    "upperLimit": "2",
  },
  {
    "id": "4bf6874c-e5ea-4f66-b2f9-e1212b823422",
    "lowerLimit": "3",
    "riskProfileDescription": "You are OK with some volatility in your portfolio.<br/>You understand that the value of your portfolio may go down for a moderate period of time before it bounces back.<br/>You understand that expected returns are average.",
    "riskProfileName": "Balanced",
    "tenantId": "0bc1e3e3-63fb-4495-beb8-411054baecda",
    "upperLimit": "3",
  },
  {
    "id": "e39f6a89-743f-4ed2-9307-22d5d545020c",
    "lowerLimit": "4",
    "riskProfileDescription": "You are OK with volatility in your portfolio.<br/>You understand that the value of your portfolio may go down for a significant period of time before it bounces back.<br/>You expect good returns in the mid to long term.",
    "riskProfileName": "Growth",
    "tenantId": "0bc1e3e3-63fb-4495-beb8-411054baecda",
    "upperLimit": "4",
  },
  {
    "id": "149a15e6-7152-4a2d-9d16-0d6a0c70b825",
    "lowerLimit": "5",
    "riskProfileDescription": "You are OK with high volatility in your portfolio.<br/>You understand that the value of your portfolio may go down sharply in the future but you know that you will reap big benefits if you are patient enough.<br/>You expect high returns in the long term.",
    "riskProfileName": "Aggressive",
    "tenantId": "0bc1e3e3-63fb-4495-beb8-411054baecda",
    "upperLimit": "5",
  },
]
`;

exports[`RiskProfiling > getRiskProfileQuesetionnaire > should return correct response 1`] = `
{
  "Questionnaire": [
    {
      "Questions": [
        {
          "Answers": [],
          "additionalConfiguration": {
            "questionFloorFlag": true,
            "questionNormalisationFactor1": "1",
            "questionNormalisationFactor2": "5",
            "scoreRangeConfig": [
              {
                "label": "<10%",
                "lowerBound": 0,
                "score": 1,
                "upperBound": 9,
              },
              {
                "answer": "10-25%",
                "lowerBound": 10,
                "score": 2,
                "upperBound": 24,
              },
              {
                "answer": "25-35%",
                "lowerBound": 25,
                "score": 3,
                "upperBound": 34,
              },
              {
                "answer": "35-50%",
                "lowerBound": 35,
                "score": 4,
                "upperBound": 49,
              },
              {
                "answer": ">50%",
                "lowerBound": 50,
                "score": 5,
                "upperBound": 100,
              },
            ],
          },
          "id": "c0b51a4b-4f17-42c9-8e63-2fa78dc52dc3",
          "question": "How much do you roughly save out of your monthly income?",
          "questionFormat": "NUMERIC_ENTRY",
          "questionType": "INCOME",
          "questionWeight": "0.5",
          "sortKey": 1,
        },
        {
          "Answers": [
            {
              "additionalConfiguration": {},
              "answer": ">85%",
              "id": "417e9e23-88b7-4563-98e5-4638dacf4154",
              "score": "1",
              "sortKey": 1,
            },
            {
              "additionalConfiguration": {},
              "answer": "<35%",
              "id": "b6659756-973e-4f2c-8163-a3991610b817",
              "score": "5",
              "sortKey": 5,
            },
            {
              "additionalConfiguration": {},
              "answer": "70-85%",
              "id": "347f5afd-28a6-469a-838a-42443b026c35",
              "score": "2",
              "sortKey": 2,
            },
            {
              "additionalConfiguration": {},
              "answer": "50-70%",
              "id": "4050301e-070d-4608-a8f8-636c3114b577",
              "score": "3",
              "sortKey": 3,
            },
            {
              "additionalConfiguration": {},
              "answer": "35-50%",
              "id": "ae152c9d-9291-4dad-bac0-cce28427a14b",
              "score": "4",
              "sortKey": 4,
            },
          ],
          "additionalConfiguration": {
            "questionFloorFlag": true,
            "questionNormalisationFactor1": "1",
            "questionNormalisationFactor2": "5",
          },
          "id": "3b216868-890a-4c47-9566-f5b6c983bd71",
          "question": "How much of your liquid assets are you planning to invest here?",
          "questionFormat": "SINGLE_CHOICE",
          "questionType": "BALANCE_SHEET",
          "questionWeight": "0.5",
          "sortKey": 2,
        },
      ],
      "additionalConfiguration": {
        "questionRoundFlag": true,
      },
      "groupingName": "FINANCIAL_HEALTH",
      "groupingType": "RISK_CAPACITY",
      "groupingWeight": "0.4",
      "id": "7ca3ad84-fa07-4bbf-8e5e-d21f5347df47",
      "scoringRules": "MAX",
      "sortKey": 1,
    },
    {
      "Questions": [
        {
          "Answers": [],
          "additionalConfiguration": {},
          "id": "b8d89946-6f33-4b77-8ea5-e49484421890",
          "question": "When do you want to achieve your goal?",
          "questionFormat": "NUMERIC_ENTRY",
          "questionType": "GOAL_TIME_FRAME",
          "questionWeight": "0",
          "sortKey": 1,
        },
      ],
      "additionalConfiguration": {
        "questionNormalisationFactor1": "1",
        "questionNormalisationFactor2": "12",
        "questionRoundDownFlag": true,
      },
      "groupingName": "GOAL",
      "groupingType": "RISK_CAPACITY",
      "groupingWeight": "0.2",
      "id": "422e5300-286f-4294-aa4b-b7adcc273bb2",
      "scoringRules": "MAX",
      "sortKey": 2,
    },
    {
      "Questions": [
        {
          "Answers": [],
          "additionalConfiguration": {},
          "id": "a624ebe3-ae95-4581-97d7-4edcae4c3f77",
          "question": "How old are you?",
          "questionFormat": "NUMERIC_ENTRY",
          "questionType": "AGE",
          "questionWeight": "0",
          "sortKey": 1,
        },
      ],
      "additionalConfiguration": {
        "questionNormalisationFactor1": "70",
        "questionNormalisationFactor2": "18",
        "questionRoundUpFlag": true,
      },
      "groupingName": "AGE",
      "groupingType": "RISK_CAPACITY",
      "groupingWeight": "0.2",
      "id": "cb5bcc9e-a911-4be0-834b-50c9e00902f8",
      "scoringRules": "MAX",
      "sortKey": 3,
    },
    {
      "Questions": [
        {
          "Answers": [
            {
              "additionalConfiguration": {},
              "answer": "Not familiar at all",
              "id": "1fd21a14-0bb5-40e6-9564-4178fb5461b8",
              "score": "1",
              "sortKey": 1,
            },
            {
              "additionalConfiguration": {},
              "answer": "Moderately familiar",
              "id": "c43a21fb-b0bf-4ff3-921d-8cc0f26cd93c",
              "score": "3",
              "sortKey": 3,
            },
            {
              "additionalConfiguration": {},
              "answer": "Somewhat familiar",
              "id": "c8f5da66-f60a-4a8d-a2f4-527addb2007f",
              "score": "2",
              "sortKey": 2,
            },
            {
              "additionalConfiguration": {},
              "answer": "Very familiar",
              "id": "d9ac8294-d708-4fda-8de2-dfbae938d3a9",
              "score": "4",
              "sortKey": 4,
            },
            {
              "additionalConfiguration": {},
              "answer": "Extremely familiar",
              "id": "51820ac7-a20f-496d-9be0-ade5d5e3cdf3",
              "score": "5",
              "sortKey": 5,
            },
          ],
          "additionalConfiguration": {},
          "id": "d5fe0921-a78f-42d7-8305-bf6014e0325e",
          "question": "How familiar are you with investing?",
          "questionFormat": "SINGLE_CHOICE",
          "questionType": "FINANCIAL_KNOWLEDGE",
          "questionWeight": "0",
          "sortKey": 1,
        },
      ],
      "additionalConfiguration": {
        "questionNormalisationFactor1": "1",
        "questionNormalisationFactor2": "5",
      },
      "groupingName": "FINANCIAL_KNOWLEDGE",
      "groupingType": "RISK_CAPACITY",
      "groupingWeight": "0.2",
      "id": "18e3b245-8138-4e01-85b8-0dd71c549e1e",
      "scoringRules": "MAX",
      "sortKey": 4,
    },
    {
      "Questions": [
        {
          "Answers": [
            {
              "additionalConfiguration": {},
              "answer": "Limited risk: this portfolio's value can go down for a short period of time but its value should be going up most of the time.",
              "id": "c35126f0-9526-43da-bab7-83d9bad6546d",
              "score": "2",
              "sortKey": 2,
            },
            {
              "additionalConfiguration": {},
              "answer": "No risk: this portfolio's value should be slowly but surely going up.",
              "id": "93d246b3-c60a-47ba-92c0-4e965e13e3e1",
              "score": "1",
              "sortKey": 1,
            },
            {
              "additionalConfiguration": {},
              "answer": "Significant risk: this portfolio's value will go down for a certain period of time but it should recover if I am patient enough.",
              "id": "42109835-a9b8-4d49-97d0-bbed07f4a916",
              "score": "3",
              "sortKey": 3,
            },
            {
              "additionalConfiguration": {},
              "answer": "High risk: this portfolio's value may fluctuate widely but I trust that it will pay off very significantly at one point in the future.",
              "id": "54808119-6022-4061-8315-84e1409d3435",
              "score": "4",
              "sortKey": 4,
            },
          ],
          "additionalConfiguration": {
            "questionNormalisationFactor1": "1",
            "questionNormalisationFactor2": "4",
          },
          "id": "c22a1a56-ed1b-47c6-bda6-461c125ac0d4",
          "question": "What level of risk are you comfortable with for this portfolio?",
          "questionFormat": "SINGLE_CHOICE",
          "questionType": "RISK_COMFORT_LEVEL",
          "questionWeight": "0.5",
          "sortKey": 1,
        },
        {
          "Answers": [
            {
              "additionalConfiguration": {},
              "answer": "I will sell a part of this portfolio to minimise exposure, but keep some in hope that prices bounce back.",
              "id": "e92ee43e-776b-44eb-83e9-c4edc43bf9cf",
              "score": "2",
              "sortKey": 2,
            },
            {
              "additionalConfiguration": {},
              "answer": "I will invest more into this portfolio since its price is likely to bounce back.",
              "id": "525edb8d-02d5-4703-b60a-94e1b0241dea",
              "score": "4",
              "sortKey": 4,
            },
            {
              "additionalConfiguration": {},
              "answer": "I will continue holding onto my portfolio and hope it returns to the original price.",
              "id": "84caa058-0c20-403b-899a-236973aca339",
              "score": "3",
              "sortKey": 3,
            },
            {
              "additionalConfiguration": {},
              "answer": "I will fully redeem this portfolio to avoid further losses.",
              "id": "84171849-70a5-4325-ae6e-e8203f62c5b3",
              "score": "1",
              "sortKey": 1,
            },
          ],
          "additionalConfiguration": {
            "questionNormalisationFactor1": "1",
            "questionNormalisationFactor2": "4",
          },
          "id": "a5b99ad4-3e58-40ad-a114-5589d54c8b85",
          "question": "Imagine you witness a sudden drop in the value of this portfolio due to market fluctuations. What will you do?",
          "questionFormat": "SINGLE_CHOICE",
          "questionType": "RISK_COMFORT_LEVEL",
          "questionWeight": "0.5",
          "sortKey": 1,
        },
      ],
      "additionalConfiguration": {
        "questionCapFlag": true,
        "questionRoundDownFlag": true,
      },
      "groupingName": "RISK_COMFORT_LEVEL",
      "groupingType": "RISK_TOLERANCE",
      "groupingWeight": "0.5",
      "id": "8bed6ec9-b81f-4190-8ba7-d4f2ed4aea97",
      "scoringRules": "MAX",
      "sortKey": 1,
    },
  ],
  "activeVersion": 1,
  "id": "cbacafa5-7d7e-4d39-9bc3-6ce1839e3266",
  "questionnaireName": "Initial Questionnaire",
  "questionnaireType": "RISK_PROFILING",
}
`;
